zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/job/JobThrowErrorProcessor.java

```diff
@@ -112,6 +115,15 @@ private void acceptCommand(
       return;
     }
 
+    // Check if the job is of kind EXECUTION_LISTENER. Execution Listener jobs should not throw
+    // BPMN errors because the element is not in an ACTIVATED state.
+    final var jobKind = job.getJobKind();
+    if (jobKind == JobKind.EXECUTION_LISTENER) {
+      commandControl.reject(
+          RejectionType.INVALID_STATE, String.format(ERROR_REJECTION_MESSAGE, jobKind));
```

Comment:
:wrench: We should provide more context to identify the job, for example, the job type and the process instance key.